packages/peregrine/lib/talons/Image/useResourceImage.js

```diff
@@ -0,0 +1,81 @@
+import { useMemo } from 'react';
+
+/**
+ * The talon for working with ResourceImages.
+ * Does all the work of generating src, srcSet, and sizes attributes.
+ *
+ * @param {func}    props.generateSrcset - A function that returns a srcSet.
+ * @param {string}  props.resource - The Magento path to the image ex: /v/d/vd12-rn_main_2.jpg
+ * @param {number}  props.resourceHeight - The height to request for the fallback image for browsers that don't support srcset / sizes.
+ * @param {Map}     props.resourceSizeBreakpoints breakpoints related to resourceSizes. Supported keys are 'small' and 'medium'.
+ * @param {Map}     props.resourceSizes image sizes used by the browser to select the image source. Supported keys are 'small', 'medium', and 'large'.
+ * @param {func}    props.resourceUrl - A function that returns the full URL for the Magento resource.
+ * @param {number}  props.resourceWidth - The width to request for the fallback image for browsers that don't support srcset / sizes.
+ * @param {string}  props.type - The Magento image type ("image-category" / "image-product"). Used to build the resource URL.
+ */
+export const useResourceImage = props => {
+    const {
+        generateSrcset,
+        resource,
+        resourceHeight,
+        resourceSizeBreakpoints,
+        resourceSizes,
+        resourceUrl,
+        resourceWidth,
+        type
+    } = props;
+
+    const src = useMemo(() => {
+        return resourceUrl(resource, {
+            type,
+            height: resourceHeight,
+            width: resourceWidth
+        });
+    }, [resource, resourceHeight, resourceUrl, resourceWidth, type]);
+
+    const srcSet = useMemo(() => {
+        return generateSrcset(resource, type);
+    }, [generateSrcset, resource, type]);
```

Comment:
It's an interesting rabbit hole, since defining inline/unstable functions in components is a known antipattern. So `generateSrcSet` should never be inline. That said, there are two reasons to put it in the deps list:

1. If `generateSrcSet` changes across renders, it should result in a de-opt, not a bug. Including it in the deps list results in a de-opt (`useMemo` creating a new object). Excluding it from the deps list would result in a bug.
2. The lint rule, `exhaustive-deps`, is not smart enough to know which functions should be stable, and we can't really hint at it (afaik). When it sees a function from props that's missing from the deps list, it complains. Disabling the rule is more destructive than including a superfluous dependency.